### PR TITLE
chore: even more linting

### PR DESCRIPTION
**Pull request by SimonSchick**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Have you added new tests to prevent regressions?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### Description of change

- Enabled eslint default rules
  - Most of the changes here are related to redundant escape sequences
- Fix JSDoc stuff

I think we're legit done after this one 😛

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -117,7 +109,23 @@
     "no-useless-call": "error",
     "no-unused-expressions": "error",
     "no-sequences": "error",
-    "no-self-compare": "error"
+    "no-self-compare": "error",
+    "no-case-declarations": "off"
+  },
+  "globals": {
+    "Association": true,
```
Comment (sushantdhiman) [inline .eslintrc.json]: This is going to be an issue with linting, now eslint will treat these as global and wont fail if someone is using `Model` without importing it first.

We can have a second linting phase for just jsdocs, with its own jsdocs related config in separate file.

Comment (SimonSchick) [reply .eslintrc.json]: Or I just change those to typedefs.

Comment (sushantdhiman) [reply .eslintrc.json]: So isn't this thread still valid. `npm run lint` wont catch misuse of these global?

Comment (SimonSchick) [reply .eslintrc.json]: I forgot to push latest changes, unfortunately type references are broken: https://github.com/gajus/eslint-plugin-jsdoc/issues/109

**Comment by SimonSchick** [comment]
@sushantdhiman apparently `test/integration/dialects/postgres/query-interface.test.js` is failing now since I've fixed the postgres version regex, should I change the regex to only match X.Y instead of X.Y.Z?

File: lib/dialects/sqlite/query-interface.js [hunk 2]
```diff
@@ -147,7 +151,8 @@ exports.removeConstraint = removeConstraint;
 
 /**
  * @param {QueryInterface} qi
- * @private
```
Comment (sushantdhiman) [inline lib/dialects/sqlite/query-interface.js]: Keep `@private`

**Comment by sushantdhiman** [comment]
> should I change the regex to only match X.Y instead of X.Y.Z?

Yes

**Pull request merged**